core/controllers/base_test.py

```diff
@@ -180,29 +180,24 @@ def test_requests_for_invalid_paths(self):
         user_id = user_services.get_user_id_from_username('learneruser')
         csrf_token = base.CsrfTokenManager.create_csrf_token(user_id)
 
-        self.get_html_response(
-            '/library/extra', expected_status_int=404)
-
         self.get_html_response(
             '/library/data/extra', expected_status_int=404)
 
         self.post_json(
-            '/library/extra', payload={}, csrf_token=csrf_token,
+            '/library/data/extra', payload={}, csrf_token=csrf_token,
             expected_status_int=404)
 
         self.put_json(
-            '/library/extra', payload={}, csrf_token=csrf_token,
+            '/library/data/extra', payload={}, csrf_token=csrf_token,
             expected_status_int=404)
 
-        self.delete_json('/library/data', expected_status_int=404)
+        self.delete_json('/library/data/extra', expected_status_int=404)
 
     def test_redirect_in_logged_out_states(self):
         """Test for a redirect in logged out state on '/'."""
 
-        response = self.get_html_response('/', expected_status_int=302)
-        response_path = response.headers['location']
-        self.assertEqual(response_path[-1], '/')
-        self.assertEqual(len(response_path.split('/')), 4)
+        response = self.get_html_response('/')
+        self.assertEqual(response.status_int, 200)
```

Comment:
Perhaps check that you eventually still land on the correct URL and that some element of the logged-out splash page is served.

Reply:
Done.